File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/ExamAttemptColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EExamAttemptStatus } from "../Web/Enums.EExamAttemptStatus";
import { ExamAttemptRow } from "./ExamAttemptRow";

export interface ExamAttemptColumns {
    Id: Column<ExamAttemptRow>;
    ExamTitle: Column<ExamAttemptRow>;
    StudentPrn: Column<ExamAttemptRow>;
    TeacherPrn: Column<ExamAttemptRow>;
    EStatus: Column<ExamAttemptRow>;
    TimeSpent: Column<ExamAttemptRow>;
    PlayListTitle: Column<ExamAttemptRow>;
    StudentAnswerSheetUpload: Column<ExamAttemptRow>;
    TeacherCheckedPaperUpload: Column<ExamAttemptRow>;
    ActivationDeviceId: Column<ExamAttemptRow>;
}

export class ExamAttemptColumns extends ColumnsBase<ExamAttemptRow> {
    static readonly columnsKey = 'Analytics.ExamAttempt';
    static readonly Fields = fieldsProxy<ExamAttemptColumns>();
}

[EExamAttemptStatus]; // referenced types

File: GXpert/GXpert.Web/Modules/Analytics/ExamAttempt/ExamAttemptColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Analytics.Columns;

[ColumnsScript("Analytics.ExamAttempt")]
[BasedOnRow(typeof(ExamAttemptRow), CheckNames = true)]
public class ExamAttemptColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ExamTitle { get; set; }
    public string StudentPrn { get; set; }
    public string TeacherPrn { get; set; }
    public short EStatus { get; set; }
    public int TimeSpent { get; set; }
    public string PlayListTitle { get; set; }
    [EditLink]
    public string StudentAnswerSheetUpload { get; set; }
    public string TeacherCheckedPaperUpload { get; set; }
    public string ActivationDeviceId { get; set; }
}